langchain/utilities/bibtex.py

```diff
@@ -0,0 +1,182 @@
+                "Please install it with `pip install bibtexparser`."
+            )
+
+        if "ignorable_exceptions" not in values:
+            values["ignorable_exceptions"] = ()
+
+        return values
+
+    def _meta_str(self, entry: Mapping[str, Any]) -> str:
+        """Return the meta information of an article in string format."""
+        metadata = self._get_metadata(entry)
+        return "\n".join([f"{key}: {value}" for key, value in metadata.items()])
+
+    def load_bibtex_entries(self, path: str) -> List[Dict[str, Any]]:
+        """Load bibtex entries from the bibtex file at the given path."""
+        import bibtexparser
+        from bibtexparser.bibtexexpression import BibtexExpression
+
+        try:
+            with open(path) as file:
+                entries = bibtexparser.load(file).entries[: self.load_max_docs]
+        except BibtexExpression.ParseException as e:
+            entries = []
+            logger.debug("Error on bibtexparser: %s", e)
+        return entries
+
+    def _get_metadata(self, entry: Mapping[str, Any]) -> Dict[str, Any]:
+        """Get metadata for the given entry."""
+        meta = {
+            "id": entry.get("ID", ""),
+            "published": entry.get("year", "Published date unknown"),
+            "title": entry.get("title", "Unknown"),
+            "publication": entry.get("journal")
+            or entry.get("booktitle")
+            or "Unknown publication",
+            "authors": entry.get("author", "Unknown authors"),
+            "summary": entry.get("abstract", "No abstract available"),
+            "url": entry.get("url", None) or f'https://doi.org/{entry.get("doi", "")}'
+            if entry.get("doi", None)
+            else "",
+        }
+        optional_fields = [
+            "annote",
+            "booktitle",
+            "editor",
+            "howpublished",
+            "journal",
+            "keywords",
+            "note",
+            "organization",
+            "publisher",
+            "school",
+            "series",
+            "type",
+            "doi",
+            "issn",
+            "isbn",
+        ]
+        if self.load_all_available_meta:
+            for field in optional_fields:
+                if field in entry:
+                    meta[field] = entry[field]
+        return meta
+
+    def run(self, file_path: str) -> str:
+        """Load bibtex file using bibtexparser and get the article meta information.
+
+        See https://bibtexparser.readthedocs.io/en/master/
+        It uses only the most informative fields of article meta information.
+        """
+        docs = [self._meta_str(entry) for entry in self.load_bibtex_entries(file_path)]
+        return (
+            "\n\n".join(docs)[: self.max_content_chars]
+            if docs
+            else "No good bibtex information found. Check your bibtex file."
+        )
+
+    def lazy_load(self, file_path: str) -> Iterator[Document]:
```

Comment:
@wavefrontshaping i added a lazy load. We want all loaders to be lazy by default. `load` invocation calls `list` on the lazy load